### PR TITLE
chore: Add cSpell dictionary and CI

### PR DESCRIPTION
**Pull request by nschonni**
- Dictionary to ignore words except for ones with open PRs to fix
- Add suggestion for vscode extension
- Add GitHub Action to run as CI
- Add problem matcher to highlight lines in PR on GitHub

### CODE REVIEW

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Spellcheck
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone repository
```
Comment (928PJY) [inline .github/workflows/spelling.yml:10]: Add the below parameters to accelerate the step
```
with:
        fetch-depth: 1
```

File: .cspell.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "vscodesessionid",
+        "vscodeversion",
+        "webpack'ed"
+    ],
+    "ignorePaths": [
+        "package-lock.json",
+        "tsconfig.json",
+        "azure-pipeline.yml",
+        ".vscode-test/**",
+        "out/**"
```
Comment (928PJY) [inline .cspell.json]:
```suggestion
        "out/**",
        "coverage/**",
        "*.vsix"
```

File: .github/workflows/spelling.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: cSpell
+        run: |
+          echo "::add-matcher::.github/workflows/cspell-problem-matcher.json"
```
Comment (928PJY) [inline .github/workflows/spelling.yml:21]: curious about how does this works? is that a feature provided by cspell or GitHub?

**Comment by 928PJY** [comment]
inspired by this file: https://github.com/w3c/specberus/blob/main/.github/workflows/js-lint.yml, maybe we should also move the lint step from our CI to a separated action to make each CI focus on less task, what is your suggestion? @nschonni

File: .github/workflows/spelling.yml (hunk 3 above)
Comment (nschonni) [reply .github/workflows/spelling.yml:21]: It's a feature of GitHub. There is a little description of them here https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md

File: .github/workflows/spelling.yml (hunk 1 above)
Comment (nschonni) [reply .github/workflows/spelling.yml:10]: Default is already 1 https://github.com/actions/checkout#usage

File: .cspell.json (hunk 2 above)
Comment (nschonni) [reply .cspell.json]: Good idea. I manually applied the change since I rebased

**Comment by nschonni** [comment]
> inspired by this file: https://github.com/w3c/specberus/blob/main/.github/workflows/js-lint.yml, maybe we should also move the lint step from our CI to a separated action to make each CI focus on less task, what is your suggestion? @nschonni

Yeah, I was going to suggest that too 😄. It might make sense to keep your other CI on Azure Pipelines, but the linting is a nice one to have run early without you having to trigger it

**Review by 928PJY — APPROVED** [review]

**Pull request merged**